### PR TITLE
HPCC-14427 Catch and log JAVA exception when loading ESDL definitions

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck anthony.fishbeck@lexisnexis.com

### CODE REVIEW

**Comment by cloLN** [comment]
@afishbeck - is this any different than what you had me test already?

**Comment by afishbeck** [comment]
@cloLN Can you run your test again and let me look at the ESP after?  My intent with this fix is to address the core, and then either document or introduce features to help address the underlying issue.  

The underlying issue in this case being when the default JVM used to compile the plugin is different then the JVM being run by ESP.  

I expect this will happen a lot and perhaps other issues will crop up... which is why I wan't to address the coring issue sooner and separately.

File: esp/services/esdl_svc_engine/esdl_binding.cpp [hunk 1]
```diff
@@ -259,8 +259,17 @@ void EsdlServiceImpl::configureJavaMethod(const char *method, IPropertyTree &ent
     if (!javaServiceMap.getValue(javaScopedClass))
     {
         VStringBuffer classPathOption("classpath=%s", classPath);
-        Owned<IEmbedServiceContext> srvctx = ensureJavaEmbeded().createServiceContext(javaScopedClass, EFimport, classPathOption);
-        javaServiceMap.setValue(javaScopedClass, srvctx.getClear());
+        try
+        {
+            Owned<IEmbedServiceContext> srvctx = ensureJavaEmbeded().createServiceContext(javaScopedClass, EFimport, classPathOption);
+            if (srvctx)
+                javaServiceMap.setValue(javaScopedClass, srvctx.getClear());
```
Comment (richardkchapman) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: Should there be an else here?

Comment (afishbeck) [inline esp/services/esdl_svc_engine/esdl_binding.cpp]: I guess I should at least log the error.  This is triggered by a change in dali so there is no return message to add an error to.

**Comment by afishbeck** [comment]
@richardkchapman Updated based on review.

**Pull request merged**